tools/export/nb/__init__.py

```diff
@@ -0,0 +1,402 @@
+        forced_includes += self.get_includes_and_remove_from_flags(flags, 'cxx_flags')
+
+        # Remove Duplicates
+        forced_includes = list(set(forced_includes))
+
+        c_std = self.get_c_std_and_remove_from_flag(flags, 'c_flags')
+        cpp_std = self.get_c_std_and_remove_from_flag(flags, 'cxx_flags')
+
+        # Make one list of all resources
+        for r_type in ['c_sources', 's_sources', 'cpp_sources']:
+            sources.extend(getattr(self.resources, r_type))
+
+        # Remove all leading './'
+        c_sources = [self.filter_dot(field) for field in getattr(self.resources, 'c_sources')]
+        cpp_sources = [self.filter_dot(field) for field in getattr(self.resources, 'cpp_sources')]
+        s_sources = [self.filter_dot(field) for field in getattr(self.resources, 's_sources')]
+        headers = [self.filter_dot(field) for field in getattr(self.resources, 'headers')]
+        sources = [self.filter_dot(field) for field in sources]
+        include_paths = [self.filter_dot(field) for field in self.resources.inc_dirs]
+
+        sys_libs = [self.prepare_sys_lib(lib) for lib
+                    in self.toolchain.sys_libs]
+        preproc = " ".join([part for part
+                            in ([basename(self.toolchain.preproc[0])] +
+                                self.toolchain.preproc[1:] +
+                                self.toolchain.ld[1:])])
+
+        if 'nbproject' in include_paths:
+            include_paths.remove('nbproject')
+
+        jinja_ctx = {
+            'name': self.project_name,
+            'target': self.toolchain.target.name,
+            'elf_location': join('BUILD', self.project_name) + '.elf',
+            'c_symbols': self.toolchain.get_symbols(),
+            'asm_symbols': self.toolchain.get_symbols(True),
+            'c_flags': flags['c_flags'],
+            'cxx_flags': flags['cxx_flags'],
+            'ld_flags': self.flags['ld_flags'],
+            'asm_flags': self.flags['asm_flags'],
+            'common_flags': self.flags['common_flags'],
+            'target': self.target,
+            'include_paths': include_paths,
+            'forced_includes': forced_includes,
+            'c_sources': c_sources,
+            'cpp_sources': cpp_sources,
+            's_sources': s_sources,
+            'headers': headers,
+            'headers_folder': self.get_netbeans_file_list(sorted(headers)),
+            'sources_folder': self.get_netbeans_file_list(sorted(sources)),
+            'options': self.options,
+            'c_std': self.get_netbeans_c_std(c_std),
+            'cpp_std': self.get_netbeans_cpp_std(cpp_std),
+            'linker_script': self.ld_script,
+            'linker_libs': sys_libs,
+            'pp_cmd': preproc
+        }
+        return jinja_ctx
+
+    def generate(self):
+        """Generate Makefile, configurations.xml & project.xml Netbeans project file
+        """
+        # super(Netbeans, self).generate()
```

Comment:
Could you remove this commented out code?

Reply:
Done!